packages/visually-hidden/src/index.tsx

```diff
@@ -11,41 +11,56 @@
  */
 
 import React, { forwardRef } from "react";
+import PropTypes from "prop-types";
+import { forwardRefWithAs } from "@reach/utils";
 
 /**
  * VisuallyHidden
  *
  * Provides text for screen readers that is visually hidden.
  * It is the logical opposite of the `aria-hidden` attribute.
  */
-const VisuallyHidden = forwardRef<
-  HTMLSpanElement,
-  React.HTMLAttributes<HTMLSpanElement>
->(function VisuallyHidden(props, ref) {
-  return (
-    <span
-      ref={ref}
-      style={{
-        border: 0,
-        clip: "rect(0 0 0 0)",
-        height: "1px",
-        margin: "-1px",
-        overflow: "hidden",
-        padding: 0,
-        position: "absolute",
-        width: "1px",
+const VisuallyHidden = forwardRefWithAs<VisuallyHiddenProps, "span">(
+  function VisuallyHidden({ as: Comp = "span", ...props }, ref) {
+    return (
+      <Comp
+        ref={ref}
+        style={{
+          border: 0,
+          clip: "rect(0 0 0 0)",
+          height: "1px",
+          margin: "-1px",
+          overflow: "hidden",
+          padding: 0,
+          position: "absolute",
+          width: "1px",
 
-        // https://medium.com/@jessebeach/beware-smushed-off-screen-accessible-text-5952a4c2cbfe
-        whiteSpace: "nowrap",
-        wordWrap: "normal",
-      }}
-      {...props}
-    />
-  );
-});
+          // https://medium.com/@jessebeach/beware-smushed-off-screen-accessible-text-5952a4c2cbfe
+          whiteSpace: "nowrap",
+          wordWrap: "normal",
```

Comment:
Wonder if we need to add `display: inline;` in this list to maintain the expected behavior for block level elements.

Reply:
I wouldn't worry about it. That said, it'd be worth spreading the `style` prop here if consumers have any issues they can solve with simple composition.

Reply:
Went ahead and took care of this for you!

Reply:
I was going to take a look at it this morning but you beat me to it. Thank you @chaance!